vm/devices/storage/disk_backend/src/lib.rs

```diff
@@ -101,8 +119,8 @@ pub trait SimpleDisk: Send + Sync + Inspect + AsyncDisk {
 
     /// Optionally returns a trait object to issue unmap (trim/discard)
     /// requests.
-    fn unmap(&self) -> Option<&dyn Unmap> {
-        None
+    fn unmap(&self) -> Option<impl Unmap> {
```

Comment:
why does `Unmap` get an impl, while other IDETs remain dyn?

Reply:
Because `Unmap` has async methods.

Reply:
Or just switching all the IDETs over to `impl`s, leaving dyn stuff to `DynDiskIo`?

Reply:
fwiw, I typically go the direction of putting the bogus impl on `core::convert::Infallible`, as opposed to defining a new type

but sure, this is OK